var/spack/repos/builtin/packages/mlpack/package.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class Mlpack(CMakePackage):
+    """mlpack is an intuitive, fast, and flexible header-only C++ machine
+    learning library with bindings to other languages. It is meant to be
+    a machine learning analog to LAPACK, and aims to implement a wide
+    array of machine learning methods and functions as a "swiss army knife"
+    for machine learning researchers."""
+
+    homepage = "https://www.mlpack.org/"
+    url = "https://github.com/mlpack/mlpack/archive/refs/tags/4.0.1.tar.gz"
+
+    maintainers("wdconinc")
+
+    version("4.1.0", sha256="e0c760baf15fd0af5601010b7cbc536e469115e9dd45f96712caa3b651b1852a")
```

Comment:
```suggestion
    version("4.2.0", sha256="f780df984a71029e62eeecdd145fb95deb71b133cefc7840de0ec706d116dd60")
    version("4.1.0", sha256="e0c760baf15fd0af5601010b7cbc536e469115e9dd45f96712caa3b651b1852a")
```
Changelog: https://github.com/mlpack/mlpack/compare/4.1.0...4.2.0

No other changes needed for 4.2.0.